ansible/roles/deploy-baremetal/tasks/prepare.yaml

```diff
@@ -0,0 +1,65 @@
+- name: Allow DHCP and HTTP services in the firewall
+  firewalld:
+    service={{item}} state=enabled
+    permanent=False immediate=True
+  with_items:
+    - http
+    - dhcp
+
+- name: Define DHCP parameters for the target node
+  template:
+    src: dhcp-host
+    dest: /etc/dnsmasq.d/{{ baremetal.ip_address }}
+    owner: root
+    group: root
+    mode: 0644
+
+# machines booting via PXE expect configuration files in some standard
+# places like:
+# pxelinux.cfg/01-ma-ca-dd-re-ss
+# pxelinux.cfg/default
+# more info: http://jk.ozlabs.org/blog/post/158/netbooting-petitboot/
+- name: Create PXE configuration file (network boot) for the target node
+  template:
+    src: pxeconfig-network
+    dest: "{{ http_server.root }}/pxelinux.cfg/01-{{ baremetal.mac_address | regex_replace(':', '-') }}"
+    owner: root
+    group: root
+    mode: 0644
+
+- name: Set machine_dir variable
+  set_fact:
+    machine_dir: "{{ http_server.root }}/{{ baremetal.mac_address }}"
```

Comment:
I dislike having things that will (almost) never change in vars.yaml because 1) people will want to change it for no reason 2) It creates confusion about which are the variables required for a specific setup and which are simply defaults.

The role cannot be a dependency of this one, because it is meant to be run only once with the controller node as a target, whereas the 'deploy' tag could be run multiple times.

Maybe we could use Ansible's group_vars to keep this more tidy. I'll explore with this a bit.